sklearn/calibration.py

```diff
@@ -175,7 +175,8 @@ def fit(self, X, y, sample_weight=None):
                     this_estimator.fit(X[train], y[train])
 
                 calibrated_classifier = _CalibratedClassifier(
-                    this_estimator, method=self.method)
+                    this_estimator, method=self.method,
+                    classes=np.unique(y[train]))
```

Comment:
@jnothman added the test (and I hope this is how you expected it to be?) 
I can make the changes which you have listed above to the class `CalibratedClassifierCV`. Do you want me to commit it in this PR or should I make another PR after this one is merged?